oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2ClientAuthenticationFilter.java

```diff
@@ -82,6 +85,9 @@ public final class OAuth2ClientAuthenticationFilter extends OncePerRequestFilter
 	private AuthenticationSuccessHandler authenticationSuccessHandler = this::onAuthenticationSuccess;
 	private AuthenticationFailureHandler authenticationFailureHandler = this::onAuthenticationFailure;
 
+	private SecurityContextHolderStrategy securityContextHolderStrategy = SecurityContextHolder
+			.getContextHolderStrategy();
+	private SecurityContextRepository securityContextRepository = new RequestAttributeSecurityContextRepository();
```

Comment:
`RequestAttributeSecurityContextRepository` does not have to be explicitly uses since `OAuth2ClientAuthenticationToken` is marked with `@Transient` and therefore will never be persisted regardless of the backing `SecurityContextRepository`